### PR TITLE
Fix valgrind nameconfig.cpp

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Let's see what travis reports.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/robotology/yarp/pull/844%23issuecomment-237205159%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23issuecomment-237208856%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23issuecomment-237221038%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23discussion_r73308800%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23discussion_r73319718%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23discussion_r73319960%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23issuecomment-237227399%22%2C%20%22https%3A//github.com/robotology/yarp/pull/844%23issuecomment-237250234%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robotology/yarp/pull/844%23issuecomment-237205159%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Actually%2C%20after%20a%20deeper%20look%20to%20the%20code%2C%20I%20suspect%20that%20%60delete%5B%5D%20ips%60%20should%20be%20called%20as%20well%20outside%20of%20the%20loop.%20The%20loop%20starting%20inside%20the%20%60%23ifdef%60%20is%20making%20me%20dizzy%2C%20I%20cannot%20really%20see%20where%20it%20is%20going.%20%3Adizzy_face%3A%3Adizzy%3A%20%22%2C%20%22created_at%22%3A%20%222016-08-03T10%3A49%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20thought%20so%20too%2C%20but%20the%20logic%20behind%20all%20the%20%60continue%60s%20etc.%20isn%27t%20too%20easy%20to%20understand%2C%20so%20I%27d%20rather%20not%20touch%20it.%22%2C%20%22created_at%22%3A%20%222016-08-03T11%3A11%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5497832%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Tobias-Fischer%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-08-03T12%3A17%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%2C%20%7B%22body%22%3A%20%22Waiting%20for%20the%20last%20builds%20to%20finish%2C%20but%20it%20seems%20like%20it%27s%20all%20fine%20now.%20Shall%20I%20create%20a%20new%20PR%20to%20merge%20this%20into%20%60master%60%3F%22%2C%20%22created_at%22%3A%20%222016-08-03T12%3A47%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5497832%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Tobias-Fischer%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hmm%20ok...%20If%20you%20want%20to%20do%20the%20things%20in%20a%20clean%20way%2C%20you%20can%20re-open%20%23843%20and%20close%20this%20pr%2C%20reset%20the%20branch%20to%20master%20and%20cherry-pick%20these%202%20commits%20%28perhaps%20they%20should%20be%20squashed%20into%20one%20commit%29%20and%20then%20force-push%20the%20branch...%20Then%20we%20will%20merge%20%23843%20into%20%60master%60%20and%20merge%20%60master%60%20into%20%60devel%60.%5Cr%5Cn%5Cr%5CnThis%20should%20do%20it%3A%5Cr%5Cn%5Cr%5Cn%60%60%60bash%5Cr%5Cn%23%20Ensure%20that%20origin/master%20is%20up%20to%20date%5Cr%5Cngit%20fetch%20origin%5Cr%5Cn%5Cr%5Cn%23%20Create%20a%20backup%20of%20the%20fix/fix-valgrind-nameconfig-2%20branch%5Cr%5Cngit%20branch%20fix/fix-valgrind-nameconfig-2-bak%20origin/fix/fix-valgrind-nameconfig-2%20%5Cr%5Cn%5Cr%5Cn%23%20Reset%20the%20local%20fix/fix-valgrind-nameconfig-2%20branch%20to%20origin/master%5Cr%5Cngit%20checkout%20fix/fix-valgrind-nameconfig-2%5Cr%5Cngit%20reset%20--hard%20origin/master%5Cr%5Cn%5Cr%5Cn%23%20Cherry%20pick%20the%202%20commits%5Cr%5Cngit%20cherry-pick%20c7eaf035c1fcaf428ef8103a90291da69d702d32%5Cr%5Cngit%20cherry-pick%20fd302953c8206b43d6facc24c3e7f975ad0c7227%5Cr%5Cn%5Cr%5Cn%23%20Merge%20the%202%20commits%20into%20one%20%5Cr%5Cngit%20reset%20HEAD%7E%5Cr%5Cngit%20add%20src/libYARP_OS/src/NameConfig.cpp%5Cr%5Cngit%20commit%20--amend%5Cr%5Cn%5Cr%5Cn%23%20Push%5Cr%5Cngit%20push%20--force%20origin%20fix/fix-valgrind-nameconfig-2%5Cr%5Cn%60%60%60%5Cr%5Cn%5Cr%5CnOtherwise%20let%27s%20just%20merge%20this%2C%20and%20I%20will%20take%20care%20of%20cherry-picking%20the%20commits...%22%2C%20%22created_at%22%3A%20%222016-08-03T14%3A19%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20c7eaf035c1fcaf428ef8103a90291da69d702d32%20src/libYARP_OS/src/NameConfig.cpp%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robotology/yarp/pull/844%23discussion_r73308800%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20%60%23endif%60%20should%20stay...%22%2C%20%22created_at%22%3A%20%222016-08-03T09%3A40%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks%20..%20fixed%20in%20https%3A//github.com/robotology/yarp/pull/844/commits/fd302953c8206b43d6facc24c3e7f975ad0c7227%22%2C%20%22created_at%22%3A%20%222016-08-03T11%3A08%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5497832%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Tobias-Fischer%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-08-03T11%3A10%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5497832%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Tobias-Fischer%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/libYARP_OS/src/NameConfig.cpp%3AL319-329%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/drdanz%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/drdanz'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/robotology/yarp/pull/844#issuecomment-237205159'>General Comment</a></b>
- <a href='https://github.com/drdanz'></a> Actually, after a deeper look to the code, I suspect that `delete[] ips` should be called as well outside of the loop. The loop starting inside the `#ifdef` is making me dizzy, I cannot really see where it is going. :dizzy_face::dizzy:
- <a href='https://github.com/Tobias-Fischer'></a> I thought so too, but the logic behind all the `continue`s etc. isn't too easy to understand, so I'd rather not touch it.
- <a href='https://github.com/drdanz'></a> :shipit:
- <a href='https://github.com/Tobias-Fischer'></a> Waiting for the last builds to finish, but it seems like it's all fine now. Shall I create a new PR to merge this into `master`?
- <a href='https://github.com/drdanz'></a> Hmm ok... If you want to do the things in a clean way, you can re-open #843 and close this pr, reset the branch to master and cherry-pick these 2 commits (perhaps they should be squashed into one commit) and then force-push the branch... Then we will merge #843 into `master` and merge `master` into `devel`.
This should do it:
```bash
# Ensure that origin/master is up to date
git fetch origin
# Create a backup of the fix/fix-valgrind-nameconfig-2 branch
git branch fix/fix-valgrind-nameconfig-2-bak origin/fix/fix-valgrind-nameconfig-2
# Reset the local fix/fix-valgrind-nameconfig-2 branch to origin/master
git checkout fix/fix-valgrind-nameconfig-2
git reset --hard origin/master
# Cherry pick the 2 commits
git cherry-pick c7eaf035c1fcaf428ef8103a90291da69d702d32
git cherry-pick fd302953c8206b43d6facc24c3e7f975ad0c7227
# Merge the 2 commits into one
git reset HEAD~
git add src/libYARP_OS/src/NameConfig.cpp
git commit --amend
# Push
git push --force origin fix/fix-valgrind-nameconfig-2
```
Otherwise let's just merge this, and I will take care of cherry-picking the commits...
- [x] <a href='#crh-comment-Pull c7eaf035c1fcaf428ef8103a90291da69d702d32 src/libYARP_OS/src/NameConfig.cpp 6'></a> &nbsp;<b><a href='https://github.com/robotology/yarp/pull/844#discussion_r73308800'>File: src/libYARP_OS/src/NameConfig.cpp:L319-329</a></b>
- <a href='https://github.com/drdanz'></a> This `#endif` should stay...
- <a href='https://github.com/Tobias-Fischer'></a> Thanks .. fixed in https://github.com/robotology/yarp/pull/844/commits/fd302953c8206b43d6facc24c3e7f975ad0c7227

<a href='https://www.codereviewhub.com/robotology/yarp/pull/844?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/robotology/yarp/pull/844?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/robotology/yarp/pull/844?approve=0'></a>&nbsp;<a href='https://github.com/robotology/yarp/pull/844'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: src/libYARP_OS/src/NameConfig.cpp [hunk 1]
```diff
@@ -319,10 +319,10 @@ ConstString NameConfig::getHostName(bool prefer_loopback, ConstString seed) {
         }
 #ifdef YARP_HAS_ACE
         delete[] ips;
-#else
-        freeifaddrs(ifaddr);
-#endif
```
Comment (drdanz) [inline src/libYARP_OS/src/NameConfig.cpp]: This `#endif` should stay...

**Comment by drdanz** [comment]
Actually, after a deeper look to the code, I suspect that `delete[] ips` should be called as well outside of the loop. The loop starting inside the `#ifdef` is making me dizzy, I cannot really see where it is going. :dizzy_face::dizzy:

File: src/libYARP_OS/src/NameConfig.cpp (hunk 1 above)
Comment (Tobias-Fischer) [inline src/libYARP_OS/src/NameConfig.cpp]: Thanks .. fixed in https://github.com/robotology/yarp/pull/844/commits/fd302953c8206b43d6facc24c3e7f975ad0c7227

Comment (Tobias-Fischer) [inline src/libYARP_OS/src/NameConfig.cpp]: :+1:

**Comment by Tobias-Fischer** [comment]
I thought so too, but the logic behind all the `continue`s etc. isn't too easy to understand, so I'd rather not touch it.

**Comment by drdanz** [comment]
:shipit:

**Comment by Tobias-Fischer** [comment]
Waiting for the last builds to finish, but it seems like it's all fine now. Shall I create a new PR to merge this into `master`?

**Comment by drdanz** [comment]
Hmm ok... If you want to do the things in a clean way, you can re-open #843 and close this pr, reset the branch to master and cherry-pick these 2 commits (perhaps they should be squashed into one commit) and then force-push the branch... Then we will merge #843 into `master` and merge `master` into `devel`.

This should do it:

```bash
# Ensure that origin/master is up to date
git fetch origin

# Create a backup of the fix/fix-valgrind-nameconfig-2 branch
git branch fix/fix-valgrind-nameconfig-2-bak origin/fix/fix-valgrind-nameconfig-2 

# Reset the local fix/fix-valgrind-nameconfig-2 branch to origin/master
git checkout fix/fix-valgrind-nameconfig-2
git reset --hard origin/master

# Cherry pick the 2 commits
git cherry-pick c7eaf035c1fcaf428ef8103a90291da69d702d32
git cherry-pick fd302953c8206b43d6facc24c3e7f975ad0c7227

# Merge the 2 commits into one 
git reset HEAD~
git add src/libYARP_OS/src/NameConfig.cpp
git commit --amend

# Push
git push --force origin fix/fix-valgrind-nameconfig-2
```

Otherwise let's just merge this, and I will take care of cherry-picking the commits...

**Comment by Tobias-Fischer** [comment]
Done. Except that it is yet another PR (https://github.com/robotology/yarp/pull/846), as a PR with a force pushed branch apparently cannot be re-opened.

**Pull request closed without merging**